ci/deploy/linux.py

```diff
@@ -58,6 +61,32 @@ def publish_deb(package: str, version: str) -> None:
     )
     bt.repo("apt").package(package).publish_uploads(version)
 
+    s3 = boto3.client("s3")
+    bucket = "materialize-apt-repository"
+    object_key = f"pool/stable/m/ma/materialized-{version}.deb"
+    # Download the staged package (deb-s3 needs to get various metadata from it)
+    s3.download_file(bucket, object_key, f"materialized-{version}.deb")
+    # Import the private key into GPG
+    key = os.environ["GPG_KEY"].replace("\\n", "\n")
```

Comment:
This seems funky. Can't we just make the environment variable contain the newlines directly?

Reply:
Perhaps, but I don't know how.

```
bash-3.2$ GPG_KEY=foo\
> bar\
> quux
bash-3.2$ echo $GPG_KEY
foobarquux
bash-3.2$ GPK_KEY="foo
> bar
> quux"
bash-3.2$ echo $GPG_KEY
foobarquux
```



Might just be that I'm lacking some basic knowledge of shell escaping.

Reply:
The last example is legit; you just need to be sure to use quotes when you print the value of the env var in bash or you won't see the embedded newlines:

```bash
GPG_KEY="foo
bar
queue"
echo "$GPG_KEY"
```

Reply:
```
bash-3.2$ GPK_KEY="foo
bar
quux"
bash-3.2$ echo "$GPG_KEY"
foobarquux
```

Reply:
You typo’d “GPG_KEY” there.

Reply:
There are bashisms that apparently let you do this, like `printf -v`, but I prefer to keep shell stuff POSIX-compliant unless necessary. No strong opinion though really.

Reply:
wow, I am stupid